### PR TITLE
Add support for JSLT

### PR DESCRIPTION
**Pull request by biochimia**
JSLT is an open-source JSON query and transformation language, inspired
by jq, XPath, and XQuery: https://github.com/schibsted/jslt.

### CODE REVIEW

File: pygments/lexers/jslt.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                'is-number', 'is-object', 'is-string', 'join', 'lowercase',
+                'max', 'min', 'mod', 'not', 'now', 'number', 'parse-time',
+                'parse-url', 'random', 'replace', 'round', 'sha256-hex', 'size',
+                'split', 'starts-with', 'string', 'sum', 'test', 'to-json',
+                'trim', 'uppercase', 'zip', 'zip-with-index', 'fallback'), suffix=_WORD_END),
+             Name.Builtin),
+            (r'[A-Z_a-z][0-9A-Z_a-z-]*:[A-Z_a-z][0-9A-Z_a-z-]*', Name.Function),
+            (r'[A-Z_a-z][0-9A-Z_a-z-]*', Name),
+            (r'\$[A-Z_a-z][0-9A-Z_a-z-]*', Name.Variable),
+            (r'.', Error),
```
Comment (Anteru) [inline pygments/lexers/jslt.py:65]: That shouldn't be needed -- unknown tokens are error tokens.

File: pygments/lexers/jslt.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+
+_WORD_END = r'(?=[^0-9A-Z_a-z-])'
+
+
+class JSLTLexer(RegexLexer):
+    """
+    For `JSLT <https://github.com/schibsted/jslt>`_ source.
+
+    .. versionadded:: 2.10.0
```
Comment (Anteru) [inline pygments/lexers/jslt.py:26]: Please change to `2.10`

**Review by Anteru — CHANGES_REQUESTED** [review]
Thakns for the contribution, I have only very small remarks.

**Comment by biochimia** [comment]
Thanks for your review, @Anteru.

I addressed both comments in https://github.com/pygments/pygments/pull/1821/commits/30fff5120c0ff842e79f4ed074117148b04e6253, which can be squashed into the first commit.

File: pygments/lexers/jslt.py (hunk 1 above)
Comment (biochimia) [reply pygments/lexers/jslt.py]: Addressed in https://github.com/pygments/pygments/pull/1821/commits/30fff5120c0ff842e79f4ed074117148b04e6253.

File: pygments/lexers/jslt.py (hunk 2 above)
Comment (biochimia) [reply pygments/lexers/jslt.py]: Addressed in https://github.com/pygments/pygments/pull/1821/commits/30fff5120c0ff842e79f4ed074117148b04e6253.

**Comment by Anteru** [comment]
Merged, thanks a lot!

**Comment by biochimia** [comment]
Thank you!

**Pull request merged**